### PR TITLE
feat: squash styles

### LINKED ISSUE #1222: Squash styles
**Do you want to request a _feature_ or report a _bug_?**

Feature

**What is the current behaviour?**

react-native-web styles are hoisted in a snapshot for comparison but are bloated by not being unique

**What is the expected behaviour?**

See [article-with-style.web.test.js.snap](https://github.com/newsuk/times-components/blob/master/packages/article/__tests__/web/__snapshots__/article-with-style.web.test.js.snap)

```
"S10": {
    "line-height": "15px",
    "margin-bottom": "0px",
    "margin-top": "0px",
    "padding-bottom": "0px",
    "padding-top": "0px",
  },
  "S11": {
    "margin-bottom": "0px",
    "margin-top": "0px",
    "padding-bottom": "5px",
    "padding-top": "5px",
  },
  "S12": {
    "line-height": "15px",
    "margin-bottom": "0px",
    "margin-top": "0px",
    "padding-bottom": "0px",
    "padding-top": "0px",
  },
```

S10 and S12 can be squashed

**If this is a feature request, what is the motivation or use case for changing
the behaviour?**

Smaller snapshots

### PR DESCRIPTION
**Pull request by craigbilner**
We have a lot of duplicate RNW styles in our snapshots, this change squashes them by removing dupes and using a single hoisted `className`

Probably best reviewed by commit:

* Provide `AppRegistry`: it's hard for external users of `jest-serializer` to use without RNW, this abstracts it to a dev dep
* Use AppRegistry: effectively the chore to implement the above without major bumping everyone
* Squash styles: the actual feature, find an existing hoisted style, use that instead
* Update snaps: the change run across all packages
* Restrict lint: now we can reduce some of the `jestlint` cheating

Close #1222

### CODE REVIEW

File: packages/author-profile/__tests__/web/__snapshots__/author-profile-with-style.web.test.js.snap [hunk 1]
```diff
@@ -79,7 +79,6 @@ exports[`1. an article list header 1`] = `
     "textDecorationLine": "none",
   },
   "S1": {},
-  "S10": {},
```
Comment (colincclark) [inline packages/author-profile/__tests__/web/__snapshots__/author-profile-with-style.web.test.js.snap]: could we go further and just remove the empties?

File: packages/jest-serializer/__tests__/web/rnw.test.js [hunk 2]
```diff
@@ -78,8 +79,41 @@ describe("The React Native Web serializer should", () => {
...
+    const component = (
+      <View style={styles.parent}>
+        <Text style={styles.child1}>child 1</Text>
+        <Text style={styles.child2}>child 2</Text>
+        <View>
+          <Text style={styles.child3}>child 3</Text>
+        </View>
+      </View>
+    );
+    const tree = renderer.create(component);
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/web/rnw.test.js]: where is that `testRenderer` lint rule when you need it?

File: packages/jest-serializer/src/rnw.js [hunk 3]
```diff
@@ -138,5 +162,5 @@ export const rnwTransform = includeStyleProps => (
   };
 };
 
-export default includeStyleProps =>
-  traverse(stylePrinter, rnwTransform(includeStyleProps));
+export default (AppRegistry, includeStyleProps) =>
```
Comment (colincclark) [inline packages/jest-serializer/src/rnw.js]: why is `AppRegistry` passed in as an arg to `rnwTransform ` and not just imported into this file?

Comment (craigbilner) [reply packages/jest-serializer/src/rnw.js]: RNW caches styles at the consumer and is not accessible by the serializer's instance. So it only works in times-components and nowhere else atm. By allowing the consumer to say, here's my RNW instance, other consumers can use the serializer.

**Pull request merged**